src/main/cpp/utils/logging/DataTraceSocket.cpp

```diff
@@ -0,0 +1,133 @@
+//====================================================================================================================================================
+/// Copyright 2024 Lake Orion Robotics FIRST Team 302
+///
+/// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+/// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+/// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+///
+/// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+/// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+/// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+/// OR OTHER DEALINGS IN THE SOFTWARE.
+//====================================================================================================================================================
+
+// FRC includes
+
+// Team 302 includes
+#include <utils/logging/DataTraceSocket.h>
+
+// Third Party Includes
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <netdb.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+
+#define PORT 30200
+
+DataTraceSocket::DataTraceSocket() : m_timer()
+{
+}
+
+void DataTraceSocket::Connect(void)
+{
+    short octets[10];
+    char host[256];
+    char *IP;
+    struct hostent *host_entry;
+    int hostname;
+    hostname = gethostname(host, sizeof(host));                      // find the host name
+    host_entry = gethostbyname(host);                                // find host information
+    IP = inet_ntoa(*((struct in_addr *)host_entry->h_addr_list[0])); // Convert into IP string
+    printf("================== Current Host Name: %s\n", host);
+    printf("================== Host IP: %s\n", IP);
+    extractIpAddress(IP, octets);
+    printf("================== Host IP(reconstructed): %d.%d.%d.%d\n", octets[0], octets[1], octets[2], octets[3]);
+
+    // since the DataTrace server is on a laptop that is getting its IP address through DHCP
+    // we need to scan a range of IP addresses starting with 10.3.2.3 to for example 10.3.2.10
+    // note that 10.3.2.1 is the compbot radio and 10.3.2.2 is the compbot roborio
+    // While this code is looking for the socket server, it will block the robot code
+    // maybe we need to find a better way
```

Comment:
Could use std::future and std::async to make this connect function asynchronous so robot code isn't blocked